object.c

```diff
@@ -4065,72 +4065,77 @@ rb_f_loop_size(VALUE self, VALUE args, VALUE eobj)
- *  classes will not be found without using a full class path.
- *
- *  A variety of strategies can be used to provide useful portions of the
- *  standard library to subclasses of BasicObject.  A subclass could
- *  <code>include Kernel</code> to obtain +puts+, +exit+, etc.  A custom
- *  Kernel-like module could be created and included or delegation can be used
- *  via #method_missing:
- *
- *    class MyObjectSystem < BasicObject
- *      DELEGATE = [:puts, :p]
- *
- *      def method_missing(name, *args, &block)
- *        return super unless DELEGATE.include? name
- *        ::Kernel.send(name, *args, &block)
- *      end
- *
- *      def respond_to_missing?(name, include_private = false)
- *        DELEGATE.include?(name)
- *      end
- *    end
- *
- *  Access to classes and modules from the Ruby standard library can be
- *  obtained in a BasicObject subclass by referencing the desired constant
- *  from the root like <code>::File</code> or <code>::Enumerator</code>.
- *  Like #method_missing, #const_missing can be used to delegate constant
- *  lookup to +Object+:
- *
- *    class MyObjectSystem < BasicObject
- *      def self.const_missing(name)
- *        ::Object.const_get(name)
+/*
+ *  Document-class: BasicObject
+ *
+ *  :markup: markdown
+ *
+ *  `BasicObject` is the parent class of all classes in Ruby.
+ *  In particular, `BasicObject` is the parent class of class Object,
+ *  which is itself the default parent class of every Ruby class:
+ *
+ *  ```
+ *  class Foo; end
+ *  Foo.superclass    # => Object
+ *  Object.superclass # => BasicObject
+ *  ```
+ *
+ *  `BasicObject` is the only class that has no parent:
+ *
+ *  ```
+ *  BasicObject.superclass # => nil
+ *  ```
+ *
+ *  \Class `BasicObject` can be used to create an object hierarchy
+ *  (e.g., class Delegator) that is independent of Ruby's object hierarchy.
+ *  Such objects:
+ *
+ *  - Do not have namespace "pollution" from the many methods
+ *    provided in class Object and its included module Kernel.
+ *  - Do not have definitions of common classes,
+ *    and so references to such common classes must be fully qualified
+ *    (`::String`, not `String`).
+ *
+ *  A variety of strategies can be used to provide useful portions
+ *  of the Standard Library in subclasses of `BasicObject`:
+ *
+ *  - The immediate subclass could `include Kernel`,
+ *    which would define methods such as `puts`, `exit`, etc.
+ *  - A custom Kernel-like module could be created and included.
+ *  - Delegation can be used via #method_missing:
+ *
+ *      ```
+ *      class MyObjectSystem < BasicObject
+ *        DELEGATE = [:puts, :p]
+ *        def method_missing(name, *args, &block)
+ *          return super unless DELEGATE.include? name
+ *          ::Kernel.send(name, *args, &block)
+ *        end
+ *        def respond_to_missing?(name, include_private = false)
+ *          DELEGATE.include?(name)
+ *        end
  *      end
- *    end
+ *      ```
```

Comment:
Can we add the newlines back between the methods? I think it improves readability.

```suggestion
 *      ```
 *      class MyObjectSystem < BasicObject
 *        DELEGATE = [:puts, :p]
 *
 *        def method_missing(name, *args, &block)
 *          return super unless DELEGATE.include? name
 *          ::Kernel.send(name, *args, &block)
 *        end
 *
 *        def respond_to_missing?(name, include_private = false)
 *          DELEGATE.include?(name)
 *        end
 *      end
 *      ```
 ```

Reply:
Done.